### ISSUE TITLE
tecio insertion of /usr/include can destroy builds

### ISSUE THREAD

**Issue by lindsayad** [libMesh/libmesh#2896]
@giudgiud recently ran into an issue where he was unable to build libmesh using conda compilers, as opposed to system compilers. He ended up with a lot of errors like
```
/home/guillaume/projects/moose/scripts/../libmesh/contrib/tecplot/tecio/tecsrc/strutil.cpp:155:46: error: 'malloc' was not declared in this scope
```
This ended up being because of the insertion of `/usr/include` in `TECIO_CPPFLAGS` which led to mixing of conda includes and system includes. It's my opinion that we should remove the autodetection of headers for tecio and any other packages to avoid these somewhat complex build failures. At least with relatively modern system compilers (or at least with my system compiler), `/usr/include` and similar directories logically associated with the system get searched automatically anyways so if the user is using a system compiler they should be fine without us helping them. And if they are *not* using a system compiler then if it turns out they can't find X11 headers or similar, leading to tecio getting disabled, then I think that's a better result than a build failure.

@jwpeterson I think you added this particular auto-detection instance. Do you have any opinions on this?

**Comment by jwpeterson** [comment]
Yeah that m4 file should at least be updated, but we could also consider dropping the tecio stuff. It's been years since that was tested, mainly because you have to have Tecplot to make sure that the files are being written correctly...
